File: CodeWars/snail.js
snail = function(array) {
  // enjoy
  let n = array[0].length;
  let fors = n + (n-1)
  let cycles = 0
  newArr = []
  while(fors > 0) {
  for ( i = cycles; i < n - cycles; i++ ) {
    newArr.push(array[cycles][i]);
  }
  fors = fors -1;
  if(fors > 1) {
    for( a = cycles + 1; a < n - cycles; a++ ){
      newArr.push(array[a][n - (cycles +1)])
    }
    fors = fors - 1;
    for( b = 1; b < n - (cycles * 2); b++ ) {
      newArr.push(array[n - (cycles + 1)][n - (cycles + 1) - b]);
    }
    fors = fors - 1;
  }
  if(fors > 1) {
    for(c = n - cycles; c > 2 + cycles; c--) {
      newArr.push(array[c-2][cycles]);
    }
    fors = fors - 1;
    cycles ++
  }
  }
  return newArr;
}

File: CodeWars/snail.py
def snail(snail_map):
    print (snail_map);
    n = len(snail_map);
    fors = n + (n-1)
    cycles = 0
    newArr = []
    while fors > 0:
        for i in range( cycles, n - cycles ):
            if len(snail_map[cycles]) > 0:
                newArr.append(snail_map[cycles][i]);
        fors = fors -1;
        if fors > 1:
            for a in range( cycles + 1, n - cycles ):
                newArr.append(snail_map[a][n - (cycles +1)])
            fors = fors - 1;
            for b in range( 1, n - (cycles * 2) ):
                newArr.append(snail_map[n - (cycles + 1)][n - (cycles + 1) - b]);
            fors = fors - 1;
        if fors > 1:
            for c in range( n - cycles, 2 + cycles, -1 ):
                newArr.append(snail_map[c-2][cycles]);
            fors = fors - 1;
            cycles = cycles + 1;
    return newArr;

    pass